eslint/configs/typescript.js

```diff
@@ -4,4 +4,15 @@ module.exports = {
 
   // Full list: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
   extends: ["plugin:@typescript-eslint/recommended"],
+
+  rules: {
+    "no-restricted-syntax": [
+      "error",
+      {
+        selector: "TSEnumDeclaration",
+        message: "Don't declare enums",
```

Comment:
Can we change message to something for descriptive of our recommendation?

Reply:
Changed it a bit, not sure about it yet tho, let me know what you think

Reply:
I think it's fine. Would maybe replace `X` with `Foo` though? Just a bit easier read I think:
```
Use `const Foo as const` with `typeof Foo` over Enum"
```